torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -283,6 +289,198 @@ inline void errorIfCapturingNonCapturableNCCL(c10::cuda::CaptureStatus status) {
 
 } // namespace
 
+namespace {
+std::string write_pickle(const c10::IValue& v) {
+  std::vector<char> result;
+  {
+    auto writer = [&](const char* data, size_t size) {
+      result.insert(result.end(), data, data + size);
+    };
+    torch::jit::Pickler pickler(
+        writer, nullptr, nullptr, nullptr, nullptr, false);
+    pickler.protocol();
+    pickler.pushIValue(v);
+    pickler.stop();
+  }
+  return std::string(result.begin(), result.end());
```

Comment:
Cool, I didn't know that C++ strings does not have to be null terminated:
https://godbolt.org/z/M7ve7dTxb